### PR TITLE
Allows promise resolver to have no arguments

### PR DESCRIPTION
**Pull request by keopiwauyu**
await generator mkae phpstan confuse i not know good fix way so i do this
https://github.com/SOF3/Capital/runs/6087223450?check_suite_focus=true#step:4:248

https://github.com/SOF3/await-generator/blob/master/await-generator/virion.yml VERSION 3.4.2

### CODE REVIEW

**Comment by keopiwauyu** [comment]
@SOF3 i edit virion.yml but you no approve& merge pr pls edit yoursel

**Comment by keopiwauyu** [comment]
why php unit and phpstan run so long very long

File: await-generator/src/SOFe/AwaitGenerator/Await.php [hunk 1]
```diff
@@ -199,7 +199,7 @@ public static function race(array $generators) : Generator{
 	 * This *differs* from JavaScript Promise in that $closure is NOT executed
 	 * until it is yielded and processed by an Await runtime.
 	 *
-	 * @param Closure(Closure(T): void, Closure(Throwable): void): void $closure
+	 * @param Closure(Closure(T): void|Closure(): void, Closure(Throwable): void): void $closure
```
Comment (SOF3) [inline await-generator/src/SOFe/AwaitGenerator/Await.php:202]:
```suggestion
	 * @param Closure(Closure(T=): void, Closure(Throwable): void): void $closure
```

**Review by SOF3 — CHANGES_REQUESTED** [review]

**Review by SOF3 — APPROVED** [review]

**Comment by SOF3** [comment]
> why php unit and phpstan run so long very long

because it requires approval.

**Comment by keopiwauyu** [comment]
> I intend to deprecate the current confusing parameter format and have something simpler like `Generator<Await, Await, Await, R>` at the phpstan level.

What ?

**Comment by SOF3** [comment]
> > I intend to deprecate the current confusing parameter format and have something simpler like `Generator<Await, Await, Await, R>` at the phpstan level.
> 
> What ?

the current API requires users to pass a recursive type into Await::all() etc. This doesn't work correctly with phpstan and users have to manually suppress the error.

**Comment by keopiwauyu** [comment]
> > > I intend to deprecate the current confusing parameter format and have something simpler like `Generator<Await, Await, Await, R>` at the phpstan level.
> > 
> > 
> > What ?
> 
> the current API requires users to pass a recursive type into Await::all() etc. This doesn't work correctly with phpstan and users have to manually suppress the error.

PHPSTAN VRRY BAD

**Comment by keopiwauyu** [comment]
@SOF3 pls pls make releaee tag i use in capital to fix phpstan err

**Pull request merged**